apache/zeppelin
# [ZEPPELIN-1210] Run interpreter per user

### What is this PR for?
Enabling each user to run same interpreter.

### What type of PR is it?
[Improvement]

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1210

### How should this be tested?
1. Enable shiro to use authentication mode
1. Check `per user` in your interpreter tab
1. Run different paragraphs with different users
  1. Run `%spark sc.version`, you will see the two `res0: ...` in your paragraphs

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No
